packages/utils/src/ast.ts

```diff
@@ -389,7 +323,43 @@ export function postProcessAST(ast: Node): Node {
       path.replaceWith(replacement as Node);
       path.skip();
     },
-  });
 
-  return ast;
+    BinaryExpression(path) {
+      const source = path.getSource();
+
+      // Throw an error if HTML comments are used as a binary expression.
+      if (source.includes('<!--') || source.includes('-->')) {
+        throw new Error('HTML comments (`<!--` and `-->`) are not allowed.');
```

Comment:
Could we elaborate on this error saying that they aren't allowed to be used as operators?

Reply:
Perhaps with a link to explain why or just a blurb that explains why